ExchangeRateApp/Controllers/CurrencyPairViewController.swift

```diff
@@ -7,116 +7,72 @@
+    var currencies: [CurrencyCodeData] = []
+    var selectedCurrencies: [CurrencyCodeData] = []
+    
     override func viewDidLoad() {
         super.viewDidLoad()
-        self.title = "Currency Pair"
         
-        if let path = Bundle.main.path(forResource: "Currencies", ofType: "json") {
-                    let dataArray = try! JSONSerialization.jsonObject(
-                            with: Data(contentsOf: URL(fileURLWithPath: path)),
-                            options: JSONSerialization.ReadingOptions()
-                    ) as! [String]
-                    
-            for rawData in dataArray {
-                let data = rawData
-                currency.append(CurrencyCodeData(code: data, fullName: "", flag: UIImage(named: "EUR")!))
+        title = "Currency Pair"
+        
+        if let path = Bundle.main.path(forResource: "Currencies", ofType: "json"),
+            let dataArray = try? JSONSerialization.jsonObject(
+                with: Data(contentsOf: URL(fileURLWithPath: path))
+            ) as? [String] {
+            
+            for data in dataArray {
+                currencies.append(CurrencyCodeData(code: data))
             }
         }
     }
-   
-    
-
-// MARK: UITableViewDataSource
-
-func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
     
-    return currency.count
-}
-
-    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+    // MARK: UITableViewDataSource
     
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        currencies.count
+    }
     
-    guard let cell = tableView.dequeueReusableCell(withIdentifier: "currencyCell", for: indexPath) as? CurrencyTableViewCell else {
-        return UITableViewCell()}
-        let item = currency[indexPath.row]
-   
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: "currencyCell", for: indexPath) as? CurrencyTableViewCell else {
+            return UITableViewCell()
+        }
+        
+        let item = currencies[indexPath.row]
+        
         cell.currencyAbriviationLabel?.text = item.code
         cell.countryLabel?.text = item.fullName.map { $0.rawValue }
         cell.countryFlagImageView.image = item.flag
         cell.countryFlagImageView.layer.cornerRadius = cell.countryFlagImageView.frame.height / 2
-        cell.countryFlagImageView.layer.masksToBounds = false
         cell.countryFlagImageView.clipsToBounds = true
         cell.countryFlagImageView.layer.borderWidth = 1
-    return cell
-
-
+        
+        return cell
     }
-
+    
     func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
-        let item = currency[indexPath.row]
-        isTapped = true
-        selectedCellPath = indexPath
-        if isTapped {
```

Comment:
`if` block doesn't make sense, since just two lines above you already set `isTapped` to `true` (it's like you put milk in the fridge, close it, open it to and check if you have milk)